valohai_yaml/objs/pipelines/override.py

```diff
@@ -69,3 +72,38 @@ def serialize(self) -> OrderedDict:  # type: ignore[type-arg]
                     elide_empty_iterables=True,
                 )
         return val
+
+    @classmethod
+    def default_merge(cls, a: Override | None, b: Any) -> Override:
+        result = copy.deepcopy(a) if a else cls()
+        result.parameters = merge_dicts(
+            b.parameters,
+            result.parameters,
+            merger=merge_simple,
+            copier=copy.deepcopy,
+        )
+        result.inputs = merge_dicts(
+            b.inputs,
+            result.inputs,
+            merger=merge_simple,
+            copier=copy.deepcopy,
+        )
+        result.environment_variables = merge_dicts(
+            b.environment_variables,
+            result.environment_variables,
+            merger=merge_simple,
+            copier=copy.deepcopy,
+        )
+        return result
+
+    def normalize_to_template(self) -> OrderedDict:
```

Comment:
This would require a docstring at the very least. It could also just modify a `template` dict passed in?